MLIR.cmake

```diff
@@ -139,7 +142,14 @@ function(add_onnx_mlir_library name)
     )
 
   if (ARG_LINK_LIBS)
-    target_link_libraries(${name} ${ARG_LINK_LIBS})
+    if (ARG_LINK_TARGET_ARCH_LIBS)
+      llvm_map_components_to_libnames(TARGET_ARCH_LIBS AllTargetsDescs AllTargetsInfos AllTargetsCodeGens AllTargetsAsmParsers MC)
+      set(TargetArchLibs)      
+      foreach(t ${TARGET_ARCH_LIBS})
+        list(APPEND TargetArchLibs "${t}")
+      endforeach(t)
+    endif()
+    target_link_libraries(${name} ${ARG_LINK_LIBS} ${TargetArchLibs})
```

Comment:
Also, ARG_LINK_LIBS might contain keywords such as private/public. By mixing the call and adding TargetArchLibs here, you're defaulting their visibility to the last keyword from ARG_LINK_LIBS. You should leave the target_link_libraries call for ARG_LINK_LIBS separate.